django/contrib/postgres/forms/jsonb.py

```diff
@@ -52,3 +52,13 @@ def prepare_value(self, value):
         if isinstance(value, InvalidJSONInput):
             return value
         return json.dumps(value)
+
+    def has_changed(self, initial, data):
+        if super().has_changed(initial, data):
+            return True
+        # For purposes of seeing whether something has changed, True is not the same as 1,
+        # and the order of keys does't matter, so they need to transform to string with same keys order.
+        data_converted = self.to_python(data)
+        initial_value = initial if initial is not None else ''
+        data_value = data_converted if data_converted is not None else ''
```

Comment:
Is this correct? You're converting `None` to `''`, but wouldn't this miss the case where the value has changed from `null` to `''`, which are distinct values?

Reply:
Thank you @orf  for review.  The converting here is because I think None and "" in json are the same value, but I am not sure is it correct?
more information(if you need):
The 'initial' parameter has called to_python by its field, so if the field' value is empty, 'initial' is None, if the field's value is "", 'initial' is u''

Reply:
None in JSON is `null` AFAIK, I think this conditional should be removed if I'm not mistaken?

Reply:
That being said, if the form value is not set and changes to 'null', this should return true as the value has changed. Maybe add `if data =='null' and initial is None: return True`?

Reply:
I think you are right,  None and "" in json are not the same value, I will fix this later.